scrunch/cubes.py

```diff
@@ -8,10 +9,21 @@ def crtabs(dataset, variables, weight=None, filter=None, **measures):
 
     :param dataset: Dataset shoji object
     :param variables: List of variable urls, names or aliases
+    :param weight: Scrunch variable instance, alias or url
+    :param filter: Scrunch expression filter
     """
-    dataset = dataset.resource
     variables = prepare_variables(variables)
-    return CrunchCube(fetch_cube(dataset, variables, count=count(), weight=weight, filter=filter, **measures))
+    if weight:
+        if isinstance(weight, Variable):
```

Comment:
We're doing the same job here, as in the funciton `prepare_variables` (checking if an argument is of type `Variable`). I'd try to expand the functionality of that function that we already have, and make it work in the case of single element arguments (or optionally make it work _only_ for single arguments, and then use a list comprehension when updating variables). I have something like this in mind:
```
variables = [prepare_variable(v) for v in variables]
weight = prepare_variable(weight)
. . .
```
It would remove the preparatory logic from the main body of the `crtabs` function, and make it more readable.